### ISSUE TITLE
CKV_AZURE_43 - false positive when using azurecaf_name resource for correct name generation

### ISSUE THREAD

**Issue by t3mi** [bridgecrewio/checkov#3511]
**Describe the issue**
CKV_AZURE_43  triggers false positive if `azurecaf_name` resource from [azurecaf provider](https://registry.terraform.io/providers/aztfmod/azurecaf/latest/docs/resources/azurecaf_name) is used for name correctness

**Examples**
```hcl
provider "azurerm" {
  features {}
}

resource "azurecaf_name" "example" {
  random_length = 20
  resource_type = "azurerm_storage_account"
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = azurecaf_name.example.result
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "GRS"

  tags = {
    environment = "staging"
  }
}
```

**Version (please complete the following information):**
 - 2.1.207

**References**
- #2924
- #2935

**Comment by JamesWoolfenden** [comment]
resolved by https://github.com/bridgecrewio/checkov/pull/4082
